device.go

```diff
@@ -178,18 +159,35 @@ func getPCIDeviceNameImpl(s *sandbox, pciID string) (string, error) {
 		case devName = <-notifyChan:
 		case <-time.After(time.Duration(timeoutHotplug) * time.Second):
 			s.Lock()
-			delete(s.deviceWatchers, pciAddr)
+			delete(s.deviceWatchers, devID)
 			s.Unlock()
 
 			return "", grpcStatus.Errorf(codes.DeadlineExceeded,
 				"Timeout reached after %ds waiting for device %s",
-				timeoutHotplug, pciAddr)
+				timeoutHotplug, devID)
 		}
 	}
 
 	return filepath.Join(systemDevPath, devName), nil
 }
 
+func getPCIDeviceNameImpl(s *sandbox, pciID string) (string, error) {
+	pciAddr, err := getDevicePCIAddress(pciID)
+	if err != nil {
+		return "", err
+	}
+
+	fieldLogger := agentLog.WithField("pciAddr", pciAddr)
+
+	// Rescan pci bus if we need to wait for a new pci device
```

Comment:
Q. How we know that we need to wait for a pci device ?

Reply:
Here just trigger a pci bus rescan, maybe the pci device has been appeared before rescan, but I think there is no harm to do an additional rescan here.